Spigot-API-Patches/0167-Add-BaseComponent-array-to-a-variety-of-events.patch

```diff
@@ -0,0 +1,2079 @@
++                        flag = RETAIN_EVERYTHING_FLAG;
++                        if ((flags & RETAIN_FLAGS) != 0) {
++                            /* Formatting flags may not be specified in combination */
++                            return -1;
++                        }
++                        break;
++                    case 'f':
++                        flag = RETAIN_FORMATTING_FLAG;
++                        if ((flags & RETAIN_FLAGS) != 0) {
++                            /* Formatting flags may not be specified in combination */
++                            return -1;
++                        }
++                        break;
++                    case 'i':
++                        flag = RETAIN_NOTHING_FLAG;
++                        if ((flags & RETAIN_FLAGS) != 0) {
++                            /* Formatting flags may not be specified in combination */
++                            return -1;
++                        }
++                        break;
++                    case 'E':
++                        flag = PASSON_EVERYTHING_FLAG;
++                        if ((flags & PASSON_FLAGS) != 0) {
++                            /* Formatting flags may not be specified in combination */
++                            return -1;
++                        }
++                        break;
++                    case 'F':
++                        flag = PASSON_FORMATTING_FLAG;
++                        if ((flags & PASSON_FLAGS) != 0) {
++                            /* Formatting flags may not be specified in combination */
++                            return -1;
++                        }
++                        break;
++                    case 'I':
++                        flag = PASSON_NOTHING_FLAG;
++                        if ((flags & PASSON_FLAGS) != 0) {
++                            /* Formatting flags may not be specified in combination */
++                            return -1;
++                        }
++                        break;
++
++                    default:
++                        /* This may not happen it's just for adding final to the flags variable */
++                        return -1;
++                }
++
++                /* Check for duplicate flags */
++                final int prevFlags = flags;
++                /* xor will remove a flag if it already exists, thus making the value smaller if it already exists */
++                if ((flags ^= flag) < prevFlags) {
++                    /* Duplicate */
++                    return -1;
++                }
++            }
++
++            if ((flags & RETAIN_FLAGS) == 0) {
++                flags |= DEFAULT_RETAIN_FLAG;
++            }
++            if ((flags & PASSON_FLAGS) == 0) {
++                flags |= DEFAULT_PASSON_FLAG;
++            }
++
++            /* Read the argument value */
++
++            final boolean readArgument = string.charAt(currentOffset) == '$';
++            int argument = -1;
```

Comment:
leave uninitialized

Reply:
nevermind, properly handle the case where an argument is specified but conversion is % or n